cli/bpmetadata/path_test.go

```diff
@@ -0,0 +1,158 @@
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := fileExists(path.Join(bptestdataPath, tt.path))
+			if (err != nil) != tt.wantErr {
+				t.Errorf("fileExists() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			if got != tt.want {
+				t.Errorf("fileExists() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func TestDirContent(t *testing.T) {
+	tests := []struct {
+		name    string
+		path    string
+		regex   string
+		want    []BlueprintMiscContent
+		wantErr bool
+	}{
+		{
+			name:  "valid examples",
+			path:  "content/examples",
+			regex: regexExamples,
+			want: []BlueprintMiscContent{
+				BlueprintMiscContent{
+					Name:     "terraform",
+					Location: "examples/acm/acm-terraform-blog-part1/terraform",
+				},
+				BlueprintMiscContent{
+					Name:     "acm-terraform-blog-part2",
+					Location: "examples/acm/acm-terraform-blog-part2",
+				},
+				BlueprintMiscContent{
+					Name:     "simple_regional",
+					Location: "examples/simple_regional",
+				},
+				BlueprintMiscContent{
+					Name:     "simple_regional_beta",
+					Location: "examples/simple_regional_beta",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:  "valid modules",
+			path:  "content/modules",
+			regex: regexModules,
+			want: []BlueprintMiscContent{
+				BlueprintMiscContent{
+					Name:     "beta-public-cluster",
+					Location: "modules/beta-public-cluster",
+				},
+				BlueprintMiscContent{
+					Name:     "binary-authorization",
+					Location: "modules/binary-authorization",
+				},
+				BlueprintMiscContent{
+					Name:     "private-cluster",
+					Location: "modules/private-cluster",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:    "invalid dir",
+			path:    "content/modules2",
+			regex:   regexModules,
+			wantErr: true,
+		},
+		{
+			name:  "some example folders without any tf",
+			path:  "content/examples-some-without-tf/examples",
+			regex: regexExamples,
+			want: []BlueprintMiscContent{
+				BlueprintMiscContent{
+					Name:     "terraform",
+					Location: "examples/acm/acm-terraform-blog-part1/terraform",
+				},
+				BlueprintMiscContent{
+					Name:     "simple_regional",
+					Location: "examples/simple_regional",
+				},
+			},
+			wantErr: false,
+		},
+		{
+			name:    "all module folders without any tf",
+			path:    "content/modules-no-tf/modules",
+			regex:   regexModules,
+			want:    []BlueprintMiscContent{},
+			wantErr: false,
+		},
+		{
+			name:    "mismatched regex",
+			path:    "content/modules",
+			regex:   "badRegex",
+			want:    []BlueprintMiscContent{},
+			wantErr: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			re := regexp.MustCompile(tt.regex)
+			got, err := getDirPaths(path.Join(bptestdataPath, tt.path), re)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("getDirPaths() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+
+			for _, wantC := range tt.want {
+				assert.Contains(t, got, wantC)
+			}
```

Comment:
Now that we sort, we can directly use equal to also test order
```suggestion
			assert.Equal(t, tt.want, got)
```

Reply:
ack